packages/core-db/src/app/ethereum/ethereum-block-processor.ts

```diff
@@ -92,6 +95,36 @@ export class EthereumBlockProcessor {
     const block: Block = await provider.getBlock(blockNumber, true)
     log.debug(`Received block: [${JSON.stringify(block)}].`)
 
+    if (
+      this.confirmsUntilFinal > 1 &&
+      !!block.transactions &&
+      !!block.transactions.length
+    ) {
+      log.debug(
+        `Waiting for ${this.confirmsUntilFinal} confirms before disseminating block ${blockNumber}`
+      )
+      // TODO: What happens on re-org? I think we're stuck waiting on this confirmation that will never come forever.
+      try {
+        await provider.waitForTransaction(
+          (block.transactions[0] as any).hash,
+          this.confirmsUntilFinal
+        )
+      } catch (e) {
+        logError(
+          log,
+          `Error waiting for ${this.confirmsUntilFinal} confirms on block ${blockNumber}`,
+          e
+        )
+        // TODO: If this is not a re-org, this may require a restart or some other action.
+        //  Monitor what actually happens and re-throw here if necessary.
```

Comment:
Correct -- I can dig into the source, but this is mainly saying that if `provider.waitForTransaction(...)` does not throw on a re-org, this will continue `await`ing, and over time, we'll consume more and more resources waiting on re-org'd transactions. That should happen infrequently enough that we're likely to bounce the service before it's an issue, though.

Separately, this is waiting on a tx because I couldn't find a good way to wait on a block. If the chain is re-org'd such that the tx is included earlier than it otherwise would be, this block will move forward with a lower-than-normal number of confirms (because this tx that was re-org'd to be part of an earlier block got N confirms, but the current block did not).

Either way, I'll re-fetch the block after the confirms come in to make sure that _if_ it got re-org'd we have the correct block & txs.

Reply:
> Either way, I'll re-fetch the block after the confirms come in to make sure that if it got re-org'd we have the correct block & txs.

Perfect that makes sense